metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -223,9 +223,17 @@ final class Doctor(
       .render
   }
 
+  private def getJdkInfo(): Option[String] = {
+    for {
+      home <- javaHome().orElse(JdkSources.defaultJavaHome)
```

Comment:
Will `.orElse(JdkSources.defaultJavaHome)` be tricky? I'm not sure if our default java detection mechanism at client side is the same as at the server-side :/

Reply:
😆 ha, I should have waited. I think for now we can put it in the header text right under the build server server. However since we keep adding it, maybe we should change the format for the json header, but let's not do that here. I can think about it a bit more and then try to figure out what is best since I'm the main person that actually uses the json ha.

Reply:
Is `ScalaTarget#jvmHome` relevant here?  Doesn't it allow a jvm home per build target or have I misunderstood its purpose

Reply:
Summarizing all aspects about java homes in Metals:
- `JdkSources.defaultJavaHome` - this java home is correct value for Metals process. 
- jdkHome from UserConfiguration. It may be different from the jdk in which the process is running and used for navigation into jdk sources.
- `ScalaTarget#jvmHome` this jvm home which bsp uses for compilation. Currently we don't use it Metals but there is an issue - #3479

Reply:
@dos65 are you sure that `UserConfiguration.javaHome` is taken into account as a second one?

Reply:
@kpodsiad in case of the current state of VSCode extension, `UserConfiguration.javaHome` and `JdkSources.defaultJavaHome` should be equal at the moment of server start. However, it's possible to start metals and then change `javaHome` in UserConfiguration and they will be different.

We anyway should have at least two fields for jdk because it will be needed after #3479.
One is current server jdk, second is that is used for sources and compilation.

Reply:
I could rename java version to `Metals Java version` to not create confusion for users but honestly, now I don't know what I should put under that field :c Maybe I should just run `System.getProperty("java.version")` + `java.vendor` + `java.version.date` without a path?